docs/storage-surface.rst

```diff
@@ -0,0 +1,916 @@
+
+To delete a bucket
+
+.. code-block:: python
+
+   >>> bucket.delete()
+
+or
+
+.. code-block:: python
+
+   >>> bucket.delete(force=True)
+
+as above.
+
+To make updates to the bucket use
+:meth:`Bucket.patch() <gcloud.storage.bucket.Bucket.patch>`
+
+.. code-block:: python
+
+   >>> bucket.versioning_enabled = True
+   >>> bucket.patch()
+
+If there are no updates to send, an exception will occur
+
+.. code-block:: python
+
+   >>> bucket.patch()
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   ValueError: No updates to send.
+
+In total, the properties that can be updated are
+
+.. code-block:: python
+
+   >>> bucket.acl = [
+   ...     ACLEntity('project-editors-111111', 'OWNER'),
+   ...     ACLEntity('project-owners-111111', 'OWNER'),
+   ...     ACLEntity('project-viewers-111111, 'READER'),
+   ...     ACLEntity('user-01234, 'OWNER'),
+   ... ]
+   >>> bucket.cors = [
+   ...     {
+   ...       'origin': ['http://example.appspot.com'],
+   ...       'responseHeader': ['Content-Type'],
+   ...       'method': ['GET', 'HEAD', 'DELETE'],
+   ...       'maxAgeSeconds': 3600,
+   ...     }
+   ... ]
+   >>> bucket.default_object_acl = [
+   ...     ACLEntity('project-owners-111111', 'OWNER'),
+   ...     ACLEntity('user-01234, 'OWNER'),
+   ... ]
+   >>> bucket.lifecycle = [
+   ...     {
+   ...         'action': {'type': 'Delete'},
+   ...         'condition': {'age': 365},
+   ...     },
+   ... ]
+   >>> bucket.location = 'ASIA'
+   >>> bucket.logging = {
+   ...     'logBucket': 'bucket-name',
+   ...     'logObjectPrefix': 'foo/',
+   ... }
+   >>> bucket.storage_class = 'DURABLE_REDUCED_AVAILABILITY'
+   >>> bucket.versioning_enabled = True
+   >>> bucket.website = {
+   ...     'mainPageSuffix': 'index.html',
+   ...     'notFoundPage': '404.html',
+   ... }
+
+In general, many of these properties are optional and will not need to be
+used (or changed from the defaults).
+
+In addition, a bucket has several read-only properties
+
+.. code-block:: python
+
+   >>> bucket.etag
+   u'CAI='
+   >>> bucket.id
+   u'bucket-name'
+   >>> bucket.metageneration
+   2L
+   >>> bucket.name
+   u'bucket-name'
+   >>> bucket.owner
+   <ACL Entity: project-owners-111111 (OWNER)>
+   >>> bucket.project_number
+   111111L
+   >>> bucket.self_link
+   u'https://www.googleapis.com/storage/v1/b/bucket-name'
+   >>> bucket.time_created
+   datetime.datetime(2015, 1, 1, 12, 0)
+
+See `buckets`_ specification for more details. `Access control`_ data is
+complex enough to be a topic of its own. We provide the
+:class:`ACLEntity <gcloud.storage.acl.ACLEntity>` class to represent these
+objects and will discuss more further on.
+
+.. _buckets: https://cloud.google.com/storage/docs/json_api/v1/buckets
+.. _Access control: https://cloud.google.com/storage/docs/access-control
+
+.. note::
+  **BREAKING THE FOURTH WALL**: Note that ``storage.buckets.update`` is
```

Comment:
If you ever offer update down the road, this will be quite confusing.  At the service level, it's really updating a specific set of metadata fields while leaving others the same.  While "update" might be semantically correct for those fields, if the operation is being applied to a bucket (or object) resource, patch is more appropriate.